llm_studio/src/utils/config_utils.py

```diff
@@ -106,7 +106,7 @@ def convert_cfg_to_nested_dictionary(cfg: ConfigProblemBase) -> dict:
 
     for k, v in cfg_dict.items():
 
-        if k.startswith("_") or cfg._get_visibility(k) < 0:
```

Comment:
What's the rationale behind removing `cfg._get_visibility(k) < 0`?

With this change, yaml config will also save non visible configuration items that are of type `KNOWN_TYPE_ANNOTATIONS`. 
I don't think it's necessarily a disadvantage to also store those fields, as they are part of the configuration.

Reply:
It is needed becasue we set `pretrained_weights` when activating `Use previous experiment weights` but it gets visibility -1 so the yaml does not store it. We can either add it explicitly here as another exception, but I thought it should not hurt to also store those not being visible.
wdyt?

Reply:
Yes, I think it's good to store all configurations, as visibility options may change.